packages/sanity/src/core/config/auth/types.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * Authentication options
+ *
+ * @public
+ */
+export interface AuthConfig {
+  /**
+   * Login method to use for the studio the studio. Can be one of:
+   * - `dual` (default) - attempt to use cookies where possible, falling back to
+   *   storing authentication token in `localStorage` otherwise
+   * - `cookie` - explicitly disable `localStorage` method, relying only on cookies. May fail due
+   *   to cookies being treated as third-party cookies in some browsers, thus the default is `dual`.
+   * - `token` - explicitly disable cookies, relying only on `localStorage` method
+   */
+  loginMethod?: 'dual' | 'cookie' | 'token'
+
+  /**
```

Comment:
Just a thought, but would it be slightly better if we typed `providers` as `Provider[] | {mode: 'append' | 'replace'; providers: Provider[]}`?

Reasoning: `mode` is a pretty broad term, and having it on the authconfig makes it seem like it's about "auth mode" (whatever that would be), so maybe more intuitive to make it part of the providers config key